plugins/python/io/NumpyReader.cpp

```diff
@@ -163,14 +210,25 @@ void NumpyReader::initialize()
     m_innersizeptr = NULL;
     m_dtype = NULL;
 
-    if (m_filename.size())
+    if (m_args->function.size() )
+    {
+        // Invoke the script and use the returned
+        // numpy array
+        //
+        m_args->source = pdal::FileUtils::readFileIntoString(m_filename);
+        m_array = load_npy_script(m_args->source,
+                                  m_args->module,
+                                  m_args->function,
+                                  m_args->fargs);
+    }
+    else if (m_filename.size())
     {
-        m_array = load_npy(m_filename);
+        m_array = load_npy_file(m_filename);
     }
-    if (m_array)
-        if (!PyArray_Check(m_array))
-            throw pdal::pdal_error("Object in file  '" + m_filename +
-                "' is not a numpy array");
+
+    if (!PyArray_Check(m_array))
```

Comment:
This error is misleading when loading from a script.  Perhaps it should be moved into the second case and a different error provided for the first.

Reply:
addressed by fddec900a5c59978842bd541e53784e578215084